src/transactions/TransactionFrame.h

```diff
@@ -312,17 +279,134 @@ class TransactionFrame : public TransactionFrameBase
     bool isSoroban() const override;
     SorobanResources const& sorobanResources() const override;
 
-    bool
-    consumeRefundableSorobanResources(uint32_t contractEventSizeBytes,
-                                      int64_t rentFee, uint32_t protocolVersion,
-                                      SorobanNetworkConfig const& sorobanConfig,
-                                      Config const& cfg);
-
     static FeePair computeSorobanResourceFee(
         uint32_t protocolVersion, SorobanResources const& txResources,
         uint32_t txSize, uint32_t eventsSize,
         SorobanNetworkConfig const& sorobanConfig, Config const& cfg);
     virtual int64 declaredSorobanResourceFee() const override;
     virtual bool XDRProvidesValidFee() const override;
+
+#ifdef BUILD_TESTS
+    friend class TransactionTestFrame;
+#endif
+};
+
+#ifdef BUILD_TESTS
+class TransactionTestFrame;
+using TransactionTestFramePtr = std::shared_ptr<TransactionTestFrame>;
+
+class TransactionTestFrame : public TransactionFrameBase
```

Comment:
Could you please add some top-level doc comment for this? Like what can we use it for and how it's different from a regular tx frame.